test/src/main/java/org/apache/accumulo/test/functional/CompactionIT.java

```diff
@@ -189,114 +170,6 @@ public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoo
-
-      assertNoCompactionMetadata(tableName);
-    }
-  }
-
-  @Test
-  public void testCompactionWithTableIterator() throws Exception {
-    String table1 = this.getUniqueNames(1)[0];
-    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
-      client.tableOperations().create(table1);
-      try (BatchWriter bw = client.createBatchWriter(table1)) {
-        for (int i = 1; i <= 4; i++) {
-          Mutation m = new Mutation(Integer.toString(i));
-          m.put("cf", "cq", new Value());
-          bw.addMutation(m);
-          bw.flush();
-          // flush often to create multiple files to compact
-          client.tableOperations().flush(table1, null, null, true);
-        }
-      }
-
-      IteratorSetting setting = new IteratorSetting(50, "delete", DevNull.class);
-      client.tableOperations().attachIterator(table1, setting, EnumSet.of(IteratorScope.majc));
-      client.tableOperations().compact(table1, new CompactionConfig().setWait(true));
-
-      try (Scanner s = client.createScanner(table1)) {
-        assertFalse(s.iterator().hasNext());
-      }
-
-      assertNoCompactionMetadata(table1);
-    }
-  }
-
-  @Test
-  public void testUserCompactionCancellation() throws Exception {
-    final String table1 = this.getUniqueNames(1)[0];
-    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
-      client.tableOperations().create(table1);
-      try (BatchWriter bw = client.createBatchWriter(table1)) {
-        for (int i = 1; i <= MAX_DATA; i++) {
-          Mutation m = new Mutation(Integer.toString(i));
-          m.put("cf", "cq", new Value());
-          bw.addMutation(m);
-          bw.flush();
-          // flush often to create multiple files to compact
-          client.tableOperations().flush(table1, null, null, true);
-        }
-      }
-
-      final AtomicReference<Exception> error = new AtomicReference<>();
-      Thread t = new Thread(() -> {
-        try {
-          IteratorSetting setting = new IteratorSetting(50, "sleepy", SlowIterator.class);
-          setting.addOption("sleepTime", "3000");
-          setting.addOption("seekSleepTime", "3000");
-          var cconf = new CompactionConfig().setWait(true).setIterators(List.of(setting));
-          client.tableOperations().compact(table1, cconf);
-        } catch (AccumuloSecurityException | TableNotFoundException | AccumuloException e) {
-          error.set(e);
-        }
-      });
-      t.start();
-      // when the compaction starts it will create a selected files column in the tablet, wait for
-      // that to happen
-      while (countTablets(table1, tm -> tm.getSelectedFiles() != null) == 0) {
-        Thread.sleep(1000);
-      }
-      client.tableOperations().cancelCompaction(table1);
-      t.join();
-      Exception e = error.get();
-      assertNotNull(e);
-      assertEquals(TableOperationsImpl.COMPACTION_CANCELED_MSG, e.getMessage());
```

Comment:
The variant of the test in ExternalCompaction_2_It does not run the compaction in backgound thread and make sure it throws an exception.

Reply:
good catch, added in 369f8d8